pygmt/src/grdcut.py

```diff
@@ -99,13 +91,13 @@ def grdcut(grid, **kwargs):
     >>> # 12° E to 15° E and a latitude range of 21° N to 24° N
     >>> new_grid = pygmt.grdcut(grid=grid, region=[12, 15, 21, 24])
     """
-    with GMTTempFile(suffix=".nc") as tmpfile:
-        with Session() as lib:
-            with lib.virtualfile_in(check_kind="raster", data=grid) as vingrd:
-                if (outgrid := kwargs.get("G")) is None:
-                    kwargs["G"] = outgrid = tmpfile.name  # output to tmpfile
-                lib.call_module(
-                    module="grdcut", args=build_arg_string(kwargs, infile=vingrd)
-                )
-
-        return load_dataarray(outgrid) if outgrid == tmpfile.name else None
+    with Session() as lib:
+        with (
+            lib.virtualfile_in(check_kind="raster", data=grid) as vingrd,
+            lib.virtualfile_out(kind="grid", fname=outgrid) as voutgrd,
+        ):
+            kwargs["G"] = voutgrd
+            lib.call_module(
+                module="grdcut", args=build_arg_string(kwargs, infile=vingrd)
+            )
+            return lib.virtualfile_to_grid(vfname=voutgrd, outgrid=outgrid)
```

Comment:
How should we handle `grdcut` prouducing either a 1-band grid or 3-band image? Should we just rename `virtualfile_to_grid` to `virtualfile_to_grid_or_image`, since the outputs will only be an `xarray.DataArray` or None?

Reply:
It's more complicated. For the virtualfile_out method, we need to pass the correct kind (grid or image) to create the virtualfile, so we need to know the output kind before calling grdcut. See the changes of grdcut in https://github.com/GenericMappingTools/pygmt/pull/3115

Reply:
Yep, so we do `grid` in, `grid` out, and `image` in, `image` out. Continue discussion at https://github.com/GenericMappingTools/pygmt/pull/3115#discussion_r1535227527